include/envoy/ratelimit/ratelimit.h

```diff
@@ -32,6 +40,19 @@ struct DescriptorEntry {
 struct Descriptor {
   std::vector<DescriptorEntry> entries_;
   absl::optional<RateLimitOverride> limit_ = absl::nullopt;
+
+  friend bool operator==(const Descriptor& lhs, const Descriptor& rhs) {
+    return lhs.entries_ == rhs.entries_ && lhs.limit_ == rhs.limit_;
+  }
+
+  // Support absl::Hash.
+  template <typename H>
+  friend H AbslHashValue(H h, const Descriptor& d) { // NOLINT(readability-identifier-naming)
+    for (const auto& entry : d.entries_) {
+      h = H::combine(std::move(h), entry.key_);
```

Comment:
Don't you have to hash on key and value here? I think it would be better to have the hash operator on DescriptorEntry if you need that. Also, remover the operator== you don't need since I don't see you hashing on all of it.

Reply:
just hashing on entries as we didn't care about match on limit here